fl4health/feature_alignment/string_columns_transformer.py

```diff
@@ -0,0 +1,23 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from fl4health.feature_alignment.constants import TextFeatureTransformer
+
+
+class StringColumnTransformer(BaseEstimator, TransformerMixin):
+    """
+    The purpose of this class is to enable the application of text feature transformers from sklearn
+    to multiple string columns, which is not supported in the first place.
+    """
+
+    def __init__(self, transformer: TextFeatureTransformer):
+        self.transformer = transformer
+
+    def fit(self, X: pd.DataFrame, y: pd.DataFrame = None) -> "StringColumnTransformer":
+        joined_X = X.apply(lambda x: " ".join(x), axis=1)
```

Comment:
If, for example, we're using a TFIDF vectorizer, doesn't this mean that the TFIDF vectors will be informed by multiple columns and represent a single text vector for all text columns concatenated together? Would it not make more sense to specify a list of transformers (one per column) and fit them to individual columns, transform those columsn and concatenate them together to produce the transformed dataframe?

Reply:
I don't mind this idea as a preprocessing option, but I'm not sure we want it to be the default. Maybe if the list of transformers is only a single transformer but receives multiple columns this can be the default behavior?